### DISCUSSION TITLE
Starting-Blocks

### DISCUSSION THREAD

**Discussion by ic360-com** [ExpressionEngine/ExpressionEngine#2972] [Feature Requests]
It would be awesome if there were some Starting Blocks available from a fresh EE install, so that people can pre-install a complete Channel and associated variables, for use with their new site. I think this would be really helpful for newbies.

Features could include:
 - Select from pre-defined types of Channel to create (Blog, News, Shop, Gallery, Events, etc).
 - Channel settings, custom fields and basic pre-built Templates to be created (probably without any styling) - in a similar way to CartThrob's "Auto-Install Templates and Channels" feature.
 - Entry Listing columns to display as relevant to the Channel type. (eg. Article Title and Date to be listed for Blog/News Entries... but Order Number, Customer Name and Order Total to be listed for Order Entries).

It would be great is these Starting-Blocks could be added to an existing site, but I appreciate it would need to have some intelligence to detect what custom field names and channel names already exist. 

[ Y ] I am capable and would like to work on implementation of this feature if it is considered.
Happy to help discuss/consult on this topic and help with design/testing.

**Comment by robsonsobral** [comment]
The Channel importer already checks for existent field groups.

It could be good to import fields and groups, besides channels. So, infinite combinations could be created.

I'm a front end dev and would gladly help on this!
